scripts/seurat_filter.R

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env Rscript
+
+if(!suppressWarnings(require(optparse))){
+    install.packages("optparse", quiet=TRUE)
+    library(optparse)
+}
+library(Seurat)
+
+#process inputs
+option_list <- list(
+  make_option(
+    opt_str = "--seurat_qc_rds",
+    type = "character",
+    help = "RDS object produced by Seurat QC"
+  ),
+  make_option(
+    opt_str = "--soupx_rds",
+    type = "character",
+    help = "RDS object produced by SoupX"
+  ),
+  make_option(
+    opt_str = "--scrublet_csv",
+    type = "character",
+    help = "CSV file with doublet information produced by Scrublet"
+  ),
+  make_option(
+    opt_str = "--output_name",
+    type = "character",
+    default = "seurat_qc.filtered.rds",
+    help = "Name for filtered Seurat RDS output"
+  )
+)
+
+opts <- parse_args(OptionParser(option_list = option_list), print_help_and_exit = TRUE)
+if (is.null(opts$seurat_qc_rds) | is.null(opts$seurat_qc_rds) | is.null(opts$soupx_rds)) {
+    write("\n\nseurat_qc_rds, soupx_rds, and scrublet_csv must be provided! One or more are missing!", stderr())
+    quit(status=1)
+}
+output_name <- opts$output_name
+seurat_qc_rds <- opts$seurat_qc_rds
+soupx_rds <- opts$soupx_rds
+scrublet_csv <- opts$scrublet_csv
+
+write("\n\nLoading Files...", stderr())
+seurat_qc_obj <- readRDS(seurat_qc_rds)
+soupx_obj <- readRDS(soupx_rds)
+scrublet_df <- read.csv(scrublet_csv, sep = ",", header = FALSE, row.names=1)
+write("Done loading files!", stderr())
+
+write("Filtering Seurat Cells...", stderr())
+write("...Begining SoupX Filtration...", stderr())
+# Report mRNA
+filter_cells <- colnames(seurat_qc_obj)[which(!(colnames(seurat_qc_obj) %in% colnames(soupx_obj)))] # Get Seurat cells not found in SoupX
+if (length(filter_cells) == 0) {
+    write("......No Seurat cells identified as mRNA contaminated. No cells removed...", stderr())
+} else {
+    write("......Cells found in Seurat were identified by SoupX as mRNA contaminated! Removing the following cells from Seurat:", stderr())
```

Comment:
Probably ok for now, may want to output summary of cells removed, etc to file in the future

Reply:
Maybe. At the moment I think it's fine since there's not many being removed but if there's a case where we have dozens or hundreds of cells removed, I can see the desire. Dunno how common that might be.